src/platform/intel/cavs/lib/pm_runtime.c

```diff
@@ -298,6 +298,53 @@ static inline void cavs_pm_runtime_core_en_memory(uint32_t index)
 #endif
 }
 
+static inline void cavs_pm_runtime_dis_dsp_pg(uint32_t index)
+{
+#if CAVS_VERSION >= CAVS_VERSION_1_8
+	uint32_t lps_ctl;
+
+	if (index == PLATFORM_MASTER_CORE_ID) {
+		lps_ctl = shim_read(SHIM_LPSCTL);
+
+		shim_write16(SHIM_PWRCTL, shim_read16(SHIM_PWRCTL) |
+			     SHIM_PWRCTL_TCPDSPPG(index) |
+			     SHIM_PWRCTL_TCPCTLPG);
+
+		lps_ctl &= ~SHIM_LPSCTL_BID;
+		lps_ctl &= ~SHIM_LPSCTL_BATTR_0;
+		lps_ctl |= SHIM_LPSCTL_FDSPRUN;
+		shim_write(SHIM_LPSCTL, lps_ctl);
+	} else {
+		shim_write16(SHIM_PWRCTL, shim_read16(SHIM_PWRCTL) |
+			     SHIM_PWRCTL_TCPDSPPG(index));
+	}
+#endif
+}
+
+static inline void cavs_pm_runtime_en_dsp_pg(uint32_t index)
+{
+#if CAVS_VERSION >= CAVS_VERSION_1_8
+	uint32_t lps_ctl;
+
+	if (index == PLATFORM_MASTER_CORE_ID) {
+		lps_ctl = shim_read(SHIM_LPSCTL);
+
+		shim_write16(SHIM_PWRCTL,
+#if CONFIG_ICELAKE
+			     SHIM_PWRCTL_TCPCTLPG);
+#else
+			     0);
+#endif
```

Comment:
I noticed the message from checkpatch, but I was waiting for a second opinion, as keeping the variable pre-proc'ed part as short as possible might improve readability.

Another alternative was to define the pwr control setting in platform specific headers but I decided to keep it here for readability since this is just an exception made for a single platform.